scvi/data/_utils.py

```diff
@@ -317,3 +317,37 @@ def _is_minified(adata: Union[AnnData, str]) -> bool:
             return uns_key in read_elem(fp["uns"]).keys()
     else:
         raise TypeError(f"Unsupported type: {type(adata)}")
+
+
+def _check_fragment_counts(
+    data: Union[pd.DataFrame, np.ndarray, sp_sparse.spmatrix, h5py.Dataset],
+    n_to_check: int = 100,
+):
+    """Approximately checks values of data to ensure it is fragment count data."""
+    # for backed anndata
+    if isinstance(data, h5py.Dataset) or isinstance(data, SparseDataset):
+        data = data[:100]
+
+    if isinstance(data, np.ndarray):
+        data = data
+    elif issubclass(type(data), sp_sparse.spmatrix):
+        data = data.data
+    elif isinstance(data, pd.DataFrame):
+        data = data.to_numpy()
```

Comment:
This line could potentially be copying the entire dataframe into a numpy. In terms of performance, it would be preferable to subset the dataframe first and then convert to numpy.